client/python/gradio_client/serializing.py

```diff
@@ -312,6 +314,7 @@ def deserialize(
 
 SERIALIZER_MAPPING = {cls.__name__: cls for cls in Serializable.__subclasses__()}
 SERIALIZER_MAPPING["Serializable"] = SimpleSerializable
+SERIALIZER_MAPPING["File"] = FileSerializable
```

Comment:
Do we need this?

Reply:
Yes I believe so. The config is returning `File` as the serializer name for File components since File components implement their own custom serialize method.

Reply:
Oh interesting, I did not know they implemented their own. Looks like the UploadButton also implements its own.

Reply:
For simplicity, can we just include the file size in `FileSerializable.serialize` as well, and then use `FileSerializable` in `File`? We'll still have to include this line for backwards compatibility but going forward it should be a little simpler

Reply:
Actually, we'll _need_ to do this. Otherwise, if you use the `File` component via API, it won't include the 'size' parameter

Reply:
Yea agreed - was just starting to think the same thing!